scheduler/filter/filter.go

```diff
@@ -63,12 +67,28 @@ func ApplyFilters(filters []Filter, config *cluster.ContainerConfig, nodes []*no
 	for _, filter := range filters {
 		candidates, err = filter.Filter(config, candidates, soft)
 		if err != nil {
-			return nil, err
+			// special case for when no healthy nodes are found
+			if filter.Name() == "health" {
+				return nil, err
+			}
+			return nil, fmt.Errorf("Unable to find a node that satisfies the following constraints %s", listAllFilters(filters, config))
```

Comment:
can we say conditions ? if we use the same terms as the code, they aren't all constraints